### PR TITLE
Hopefully fix directory assumptions of deploy script

### PR DESCRIPTION
**Pull request by DRMacIver**
As expected, #1210 didn't entirely work on merge. This hopefully fixes that. I was going to push straight to master but it ended up a bit larger than I intended (though still not large), so PR after all.

Unfortunately very hard to test without actually running it. We definitely need some testing of our build. :-(

### CODE REVIEW

File: scripts/hypothesistooling.py [hunk 1]
```diff
@@ -404,3 +405,23 @@ def should_run_ci_task(task, is_pull_request):
...
+SECRETS_TAR = SECRETS_BASE + '.tar'
+ENCRYPTED_SECRETS = SECRETS_TAR + '.enc'
+
+DEPLOY_KEY = os.path.join(ROOT, 'deploy_key')
+PYPIRC = os.path.join(ROOT, '.pypirc')
+
+
+def decrypt_secrets():
+    subprocess.check_call([
+        'openssl', 'aes-256-cbc', '-K', '$encrypted_39cb4cc39a80_key',
```
Comment (alexwlchan) [inline scripts/hypothesistooling.py]: Do you want `os.environ['encrypted_39cb4cc39a80_key']`? Shell variables aren’t going to be expanded here, are they?

Comment (DRMacIver) [reply scripts/hypothesistooling.py]: Oh! Good point. Thank you.

**Comment by DRMacIver** [comment]
@alexwlchan I've also added a Travis task that checks that secret decryption is working correctly.

File: Makefile [hunk 2]
```diff
@@ -285,3 +285,6 @@ doctest: $(SPHINX_BUILD)
 
 fix_doctests: $(TOOL_VIRTUALENV)
 	PYTHONPATH=src $(TOOL_PYTHON) scripts/fix_doctests.py
+
+check_secrets: $(TOOL_VIRTUALENV)
```
Comment (alexwlchan) [inline Makefile]: Underscore != hyphen

File: scripts/check_secrets.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import os
+import sys
+
+import hypothesistooling as tools
+
+if __name__ == '__main__':
+    if os.environ.get('TRAVIS_SECURE_ENV_VARS', None) != 'true':
+        sys.exit(0)
+
+    tools.decrypt_secrets()
```
Comment (alexwlchan) [inline scripts/check_secrets.py]: Maybe add an assert that this does something useful? e.g.

```python
assert not os.path.exists(tools.SECRETS_BASE)
tools.decrypt_secrets()
assert os.path.exists(tools.SECRETS_BASE)
```

File: Makefile (hunk 2 above)
Comment (DRMacIver) [reply Makefile]: Oops. Thanks.

File: scripts/check_secrets.py (hunk 3 above)
Comment (DRMacIver) [reply scripts/check_secrets.py]: SECRETS_BASE actually doesn't exist - we unpack everything into the root (this is probably a bad idea, but fixing it is out of scope for this PR... I'll look into more sensible secret management when I merge in hypothesis-ruby)

**Pull request merged**